### PR TITLE
initial smart contract dev setup

### PR DESCRIPTION
**Pull request by JamesEarle**
Add initial setup for smart contract dev with required dependencies.

Include `logdown` for logging as well as `prettier` for formatting. In VS Code, you can set to format on every save by navigating to `File > Preferences > Settings` and searching for "format on save"

This requires having a formatter, so make sure you install all the dependencies with yarn first.

image

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        "printWidth": 80,
+        "tabWidth": 2,
+        "useTabs": false,
+        "singleQuote": false,
+        "bracketSpacing": false,
+        "explicitTypes": "preserve"
+      }
+    },
+    {
+      "files": "*.ts",
```
Comment (Will-Hyde) [inline .prettierrc:15]: is it worth it to run prettier on .json also?

File: contracts/Registry.sol [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
```
Comment (Will-Hyde) [inline contracts/Registry.sol:2]: Should we lock the version? Or are we comfortable with allowing  it to upgrade to minor releases?

**Review by Will-Hyde — APPROVED** [review]

File: contracts/Registry.sol (hunk 2 above)
Comment (Whytecrowe) [reply contracts/Registry.sol:2]: should be fine, but I like locking as well

File: .prettierrc (hunk 1 above)
Comment (Whytecrowe) [reply .prettierrc:15]: we can add it later if needed.

**Review by Whytecrowe — APPROVED** [review]

File: contracts/Registry.sol (hunk 2 above)
Comment (JamesEarle) [reply contracts/Registry.sol:2]: I will lock the version

**Pull request merged**